Update README.md

ryutaogiwara opened the pull request:
# WHAT
README.mdファイルにデータベース設計に関するマークダウンを作成
# WHY
データベースの構造に間違いがないか精査するため

ryutaogiwara committed:
Create README.MD

WHAT
app/README.mdにデータベース設計を記載したマークダウン作成
WHY
データベース設計をするため

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
+|id|integer|null: false|
+|name|string|null: false, unipue: true|
+|mail_address|string|null: false, unique: true|
 
```
パスワードも必要です。

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
 
-Things you may want to cover:
+### Association
+- belongs_to :group
```
複数のgroupに入ることがあるかと思います。

ryutaogiwara replied:
アソシエーション追加しました

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
-* Configuration
+|Column|Type|Options|
+|------|----|-------|
+|user_id|integer|null: false, foreign_key: true|
```
integerでは外部キー制約にならないため、reference型にしましょう。

ryutaogiwara replied:
reference型に変更しました

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
-* Services (job queues, cache servers, search engines, etc.)
+|Column|Type|Options|
+|------|----|-------|
+|id|integer|null: false|
```
IDは自動的に作成されるため削除しましょう。

ryutaogiwara replied:
ID columns削除しました

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
-* Deployment instructions
+### Association
+- has_many   :users
+
```
メッセージもグループに紐づける必要があります。また中間テーブルへのアソシエーションも必要です。

ryutaogiwara replied:
アソシエーション見直しました

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
+
+|Column|Type|Options|
+|------|----|-------|
+|id|integer|null: false|
```
ID削除しましょう。

ryutaogiwara replied:
id columns　削除しました

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
+|Column|Type|Options|
+|------|----|-------|
+|id|integer|null: false|
+|body|string|null: false|
```
テキストだけ、画像だけでも投稿することが考えられるので、null: falseは必要ありません。

ryutaogiwara replied:
カラムオプション修正しました

ato-tech commented on README.md:
```diff
@@ -1,24 +1,50 @@
+
+### Association
+- belongs_to :user
 
```
グループとも紐づける必要があります。

ryutaogiwara replied:
アソシエーション修正しました

ryutaogiwara committed:
Update README.md

・各アソシエーション見直し
・groups_usersテーブルをmenbersテーブルへ変更
・menbersテーブルをリファレンス型へ変更
・自動生成されるidカラムについては削除

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,51 @@
-* Ruby version
 
-* System dependencies
+## menbers table
```
membersですね

ryutaogiwara replied:
修正しました

TOMARUMARU commented on README.md:
```diff
@@ -1,24 +1,51 @@
+|------|----|-------|
+|user_id|reference|null:false, foreign_key: true|
+|group_id|reference|null:false, foreign_key: true|
+|message_id|reference|foreign_key: true|
```
memberにはmessgeはひもづきません

ryutaogiwara replied:
修正しました

TOMARUMARU reviewed (commented):
groupsのアソシエーションが不十分です。中間テーブルを用いたアソシエーションについて調べて修正をお願いします

ryutaogiwara committed:
Update README.md

中間テーブルを用いたアソシエーションを見直しました。

TOMARUMARU reviewed (approved):
LGTM
